validator/lib/readTileset.js

```diff
@@ -0,0 +1,15 @@
+'use strict';
+var fs = require('fs-extra');
+var Promise = require('bluebird');
+
+var fsReadFile = Promise.promisify(fs.readJson);
+
+function readTileset(filePath) {
```

Comment:
You will have to rework this function to support checking if the file is gzipped or not. Check out `readTile` for an example.